app/src/main/java/org/kinecosystem/kinit/util/GeneralUtils.java

```diff
@@ -9,13 +9,26 @@
 import android.net.NetworkInfo;
 import android.net.Uri;
 import android.os.Build;
+import android.support.annotation.Nullable;
 import android.text.TextUtils;
 import android.util.DisplayMetrics;
 import android.view.View;
 import android.view.inputmethod.InputMethodManager;
 
+import org.json.JSONException;
+import org.json.JSONObject;
+
 public class GeneralUtils {
 
+    public static @Nullable JSONObject stringToJson(String jsonStr){
```

Comment:
Instead of adding this method - you can create specific data classes for all type of json objects that you expect from javascript and then use Gson to convert the json string to the specific data class object. 

see example in KinMessagingService.java 
```
Gson gson = new Gson()
AuthTokenMessage authTokenMessage = gson.fromJson(message, AuthTokenMessage.class);
```
message is a String in json format.

Reply:
I read about it - and thought it's an overkill.